http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -30,15 +30,13 @@ sealed trait ContainerRegistry extends Product with Serializable {
 }
 object ContainerRegistry {
   final case object GCR extends ContainerRegistry {
-    def regex: Regex = """(.*gcr.io)\/(.+)\/(.+)(\:?.*)$""".r
-
+    val regex: Regex = """^(.*gcr.io)/([\w.-]+/[\w.-]+)(?::(\w[\w.-]+))?(?:@([\w+.-]+:[A-Fa-f0-9]{32,}))?$""".r
```

Comment:
Tweaked this regex to be more like the GCR one

Reply:
Should we be more strict to ensure expressions would only match one of the four possible `gcr.io`s?

```suggestion
    val regex: Regex = """^((us\.|eu\.|asia\.)?gcr.io)/([\w.-]+/[\w.-]+)(?::(\w[\w.-]+))?(?:@([\w+.-]+:[A-Fa-f0-9]{32,}))?$""".r
```

Reply:
It could be nice to add negative tests against paths starting with `.gcr.io`, `foogcr.io`, `foo.gcr.io`, `foo.bar.gcr.io`, `foo.gcr.io.bar`, etc.

Reply:
Yeah that's reasonable, although I think the suggestion is not quite right (there's an unmatched paren). I'll try this and improve the regex unit tests.

Reply:
Ah yea...I've updated it. Hope it works now.

Reply:
It does! added in latest commit